test/types.test.ts

```diff
@@ -189,3 +189,160 @@ test(`undefined isn't removed from unions`, () => {
   expectTypeOf<string | null | undefined>().toEqualTypeOf<string | null | undefined>()
   expectTypeOf<string | null | undefined>().toMatchTypeOf<string | null | undefined>()
 })
+
+// Note: Works because of DeepBrand `type: 'function'` properties `params` and `return`.
+test('Distinguish between functions whose return types differ by readonly prop', () => {
```

Comment:
This test case was passing before my changes to `index.ts`

All other test cases do not pass without the changes to `index.ts`